aws-analytics-pinpoint/src/test/java/com/amplifyframework/analytics/pinpoint/AutoSessionTrackerTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics.pinpoint;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import com.amazonaws.mobileconnectors.pinpoint.analytics.AnalyticsClient;
+import com.amazonaws.mobileconnectors.pinpoint.analytics.SessionClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests the {@link AutoSessionTracker}.
+ */
+@RunWith(RobolectricTestRunner.class)
+public final class AutoSessionTrackerTest {
```

Comment:
For some of these, you have "app is in goreground" was a "when" step, but it is more like a "given."

For example, "Given that the app had already been opened, when the user hits the home button, then the session is stopped."

This will change your "then" logic slightly too, since I think you'll only want to verify `stopSession()`, in that case, not _both_ `startSession()` and `stopSession()`, back to back.

Although, that `InOrder` is cool!! I'm surprised I've never seen it before. Nice find.